src/nbl/asset/utils/CSPIRVIntrospector.cpp

```diff
@@ -109,20 +109,127 @@ static CSPIRVIntrospector::CStageIntrospectionData::VAR_TYPE spvcrossType2E_TYPE
 // returns true if successfully added all the info to self, false if incompatible with what's already in our pipeline or incomplete (e.g. missing spec constants)
 NBL_API2 bool CSPIRVIntrospector::CPipelineIntrospectionData::merge(const CSPIRVIntrospector::CStageIntrospectionData* stageData, const ICPUShader::SSpecInfoBase::spec_constant_map_t* specConstants)
 {
-    return false;
+
+    // merge descriptors
+    for (uint32_t i = 0u; i < ICPUPipelineLayout::DESCRIPTOR_SET_COUNT; i++)
+    {
+        std::ostringstream debug;
+        debug << "ds ID: " << i << std::endl;
+        const auto& bindingInfos = stageData->getDescriptorSetInfo(i);
+        for (const auto& bindingInfo : bindingInfos)
+        {
+            CPipelineIntrospectionData::SDescriptorInfo descInfo;
+            descInfo.binding = bindingInfo.binding;
+            descInfo.type = bindingInfo.type;
+            
```

Comment:
you need to get the array-stride and the count.

Getting the count is mildly complicated:
1. if the `bindingInfo.count.isRuntimeSized()` then your `descInfo` needs to be runtime sized as well
2. else if the `bindingInfo.count.isSpecConstant` you need to look up the value in the `specConstant` map (if its nullptr, you need to return false and fail now) and substitute that for the `descInfo.count`
3. otherwise you're hitting the plain regular 99% of the time path, which is `!isSpecConstant` and you just take the `count.value` to become the `descInfo.count`

Reply:
ah one more thing about (1), only the very last binding is allowed to be Runtime Sized (counterpart to variables in an SSBO but for descriptors) so if you have a runtime sized `bindingInfo` you need to check if its the last one in the `bindingInfos` and if its not, return false.

Reply:
(2) was in the old code by Criss